### ISSUE TITLE
proposal: remove the postinstall script

### ISSUE THREAD

**Issue by bengl** [ForbesLindesay/spawn-sync#25]
As seen with #11 and others, the `postinstall` script is prone to issues. In the case of #11, it's because `npm` drops permissions before running the lifecycle scripts, so if `sudo` was used to install `spawn-sync` (as is the case when many people install `yo`, which depends on `spawn-sync`), the install errors out.

Since `spawn-sync` is probably most useful in CLI-tools, and unfortunately, many CLI-tools are designed to be installed with `-g`, this problem is bound to keep coming up.

The solutions provided on the ticket (`--unsafe-perms` and using a node version manager so that things are installed in user directories and sudo isn't needed) aren't really ideal. Let's just nuke it and install all the dependencies all the time. Maybe there's a better solution, but I think it's worth backpedalling here to not break installs.

In short, I'm not sure saving some time in install is worth breaking the install for piles and piles of `yo` and other CLI-tool users.

**Comment by ForbesLindesay** [comment]
The issue with removing the postinstall script is that it makes the experience worse for all users of 0.12 or later.  In the long run, that will be the vast majority of users.  By contrast, having the postinstall script will only cause problems for users who are on an old version of node.js, with incorrectly configured permissions.  i.e. it is a short term problem.

I've added more try/catch blocks which should catch the vast majority of postinstall script issues (with users falling back to just busy waiting without the help of the thread-sleep module).  I will wait and see if anyone still has problems.
